File: 2nd_Course/Q1/POO/Unit_1/Unit_1_Practice/src/ex03/ex03.java
package ex03;

/*
 * Escribir un programa que calcule el mínimo, el máximo 
 * y la media de una lista de números enteros positivos introducidos por el usuario. 
 * La lista finalizará cuando se introduzca un número negativo.
 * 
 * */
import java.util.Scanner;

public class ex03 {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in); 
		int numero, maximo, minimo, suma = 0, cantidad = 0; 
		
		System.out.println("introduce un numero positivo: ");
		numero = scanner.nextInt(); 
		minimo = maximo = numero; 
		
		while(numero >= 0) {
			suma += numero; 
			cantidad++; 
			if(numero < minimo)
				minimo = numero; 
			if(numero > maximo)
				maximo = numero; 
			System.out.println("introduce otro numero positivo para continuar o negativo si quieres terminar: ");
			numero = scanner.nextInt(); 	
		}
		if (cantidad > 0) {
			double media = (double) suma / cantidad; 
			System.out.println("Mínimo: " + minimo);
			System.out.println("Maximo: "+ maximo);
			System.out.println("Media: " + media);
		} else {
			System.out.println("no se introdujeron numeros positivos!");
		}
	}
}

File: 1st_Course/Q1/Programming_1/U3/Extra_Exercises/ex03.c
#include <stdio.h>

int esPerfecto(int numero);

int main()
{
    for (int i = 1; i <= 10000; i++)
    {
        if (esPerfecto(i))
        {
            printf("%d es un número perfecto\n", i);
        }
    }
    return 0;
}

int esPerfecto(int numero)
{
    int suma = 0;
    for (int i = 1; i < numero; i++)
    {
        if (numero % i == 0)
        {
            suma += i;
        }
    }
    return suma == numero;
}
